src/objects/BatchedMesh.js

```diff
@@ -899,7 +900,7 @@ class BatchedMesh extends Mesh {
 		const drawRanges = this._drawRanges;
 		const perObjectFrustumCulled = this.perObjectFrustumCulled;
 
-		// prepare the frustum
+		// prepare the frustum in the local frame
 		if ( perObjectFrustumCulled ) {
 
 			_projScreenMatrix
```

Comment:
It occurred to me that I don't fully understand how this frustum generation logic works - it was copied from WebGLRenderer and then ".muliply( this.matrixWorld )" was added - why is it that we use camera.matrixWorldInverse here instead of camera.matrixWorld to get the frustum in world frame?

Reply:
In `WebGLRenderer`, `_projScreenMatrix` transforms from world space directly to clip space and maybe it would be less confusing to stick to logic in `BatchedMesh`, too.

Meaning `.multiply( this.matrixWorld );` should be removed and `_frustum` needs to check things in world space.

Reply:
That's significantly more matrix multiplications for no gain, though, since it means each bounding volume needs to have the world matrix multiplied in.

I just want to understand how the `Frustum.setFromProjectionMatrix` function works. It must take an inverted matrix? It must be true since `camera.projectionMatrix` is what compresses everything into the clip space box.  Even if we just talk about what WebGLRenderer is doing - it wasn't clear why this results in a frustum that can check for intersections in world space since `camera.matrixWorldInverse` should transform into the _local_ space of the camera.

```js
_projScreenMatrix.multiplyMatrices( camera.projectionMatrix, camera.matrixWorldInverse )
```